### ISSUE TITLE
NativeImage does not adapt to retina display

### ISSUE THREAD

**Issue by erikjalevik** [electron/electron#17784]
### Preflight Checklist
* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 4.1.4
* **Operating System:**
  * MacOS 10.14.4
* **Last Known Working Electron version:**:
  * n/a

### Expected Behavior
The [documentation for nativeImage](https://electronjs.org/docs/api/native-image#high-resolution-image) states that "you can put images with different sizes in the same folder and use the filename without DPI suffixes" to support displays with different DPIs. I have prepared files `name.png` and `name@2x.png`. In the code I am using just `name.png` to refer to it. I expected it to use the `@2x` version of a PNG when displaying on a Macbook retina screen, and the version without a suffix when displaying on an older external non-retina monitor.

### Actual Behavior
The low-resolution version of the image is always used when specifying the filename without a suffix.

### To Reproduce
Here is a repo showing the issue: 

https://github.com/erikjalevik/electron-retina-nativeimage-bugs

Instructions:
```
git clone git@github.com:erikjalevik/electron-retina-nativeimage-bugs.git
npm install
npm run start
```

When running this on a retina screen, I expect the first image in each section "Adaptive image" to be the same as the third image "Hardcoded 2x". When on non-retina, I expect the first image to be the same as the second image "Hardcoded 1x".

In short:
1. Prepare a 1x and 2x version of a PNG
2. Use `nativeImage.createFromPath` passing the filename without a suffix.
3. Use the returned image's `toDataURL()` and set its return value to the `src` attribute of an `<img>` tag.

NB. The same issue is also present when using `app.getFileIcon` as shown in the second section of the repo's demo page.

### Screenshots

<img width="438" alt="Screenshot 2019-04-15 at 13 45 04" src="https://user-images.githubusercontent.com/8722592/56130365-f7d89980-5f84-11e9-842a-2d95785bf8bc.png">

**Comment by miniak** [comment]
@erikjalevik `nativeImage.createFromPath` works property. The problem is that `toDataURL()` uses 1.0 as the default `scaleFactor`, unless specified in the `options`.

**Comment by miniak** [comment]
you have to do something like this `document.getElementById("folder").src = adaptive.toDataURL({ scaleFactor: devicePixelRatio });`

**Comment by erikjalevik** [comment]
Thanks for clearing that up, @miniak!

However, now that I'm trying it out, it seems that I can only get the device pixel ratio by asking for a specific display from `screen` and then call `scaleFactor` on that. Then somehow listen for the app being moved about and detect when it's moving on/off retina screens and then re-render the icons?

Based on the `nativeImage` docs, I got the impression that it would be automatically handled based on the file suffix.
